mesos/src/main/java/brooklyn/entity/mesos/framework/marathon/MarathonFrameworkImpl.java

```diff
@@ -185,61 +191,42 @@ public void start(Collection<? extends Location> locs) {
      */
     @Override
     public MarathonLocation createLocation(Map<String, ?> flags) {
-        String locationName = MarathonResolver.MARATHON + "-" + getId();
-        String locationSpec = String.format(MarathonResolver.MARATHON_FRAMEWORK_SPEC, getId());
-        sensors().set(LOCATION_SPEC, locationSpec);
-
-        LocationDefinition definition = new BasicLocationDefinition(locationName, locationSpec, flags);
-        Location location = getManagementContext().getLocationRegistry().resolve(definition);
-        getManagementContext().getLocationRegistry().updateDefinedLocation(definition);
-
-        ManagementContext.PropertiesReloadListener listener = DockerUtils.reloadLocationListener(getManagementContext(), definition);
-        getManagementContext().addPropertiesReloadListener(listener);
-        sensors().set(Attributes.PROPERTIES_RELOAD_LISTENER, listener);
-
-        sensors().set(LocationOwner.LOCATION_DEFINITION, definition);
-        sensors().set(LocationOwner.DYNAMIC_LOCATION, location);
-        sensors().set(LocationOwner.LOCATION_NAME, location.getId());
-
-        LOG.info("New Marathon framework location {} created", location);
-        return (MarathonLocation) location;
-    }
-
-    @Override
-    public void rebind() {
-        super.rebind();
-
-        // Reload our location definition on rebind
-        ManagementContext.PropertiesReloadListener listener = sensors().get(Attributes.PROPERTIES_RELOAD_LISTENER);
-        if (listener != null) {
-            listener.reloaded();
+        String locationName = config().get(LOCATION_NAME);
```

Comment:
This will always get the parent configuration, so creating the location will fail. I have a fix that replicates the naming logic that was there originally, which I will commit after merging these changes.